sys/kern/kern_cheri_revoke.c

```diff
@@ -95,6 +95,57 @@ cheri_revoke_fini(struct cheri_revoke_syscall_info * __capability crsi,
 	return (res3);
 }
 
+static int
+sysctl_kern_proc_quarantining(SYSCTL_HANDLER_ARGS)
+{
+	struct vmspace *vm;
+	int *name = (int *)arg1;
+	u_int namelen = arg2;
+	struct proc *p;
+	int error;
+	pid_t pid;
+	int is_quarantining;
+
+	if (namelen != 1)
+		return (EINVAL);
+
+	pid = (pid_t)name[0];
+	p = curproc;
+	if (pid == p->p_pid || pid == 0) {
+		if (!SV_CURPROC_FLAG(SV_CHERI)) {
+			is_quarantining = -1;
+			goto out_nocheri;
+		}
+
+		vm = vmspace_acquire_ref(p);
```

Comment:
You don't really need this ref. If you stop taking it, you can combine the `out` and `out_nocheri` labels.